### PR TITLE
Technote for code development in notebooks

### PR DESCRIPTION
**Pull request by SimonKrughoff**

### CODE REVIEW

File: index.rst [hunk 1]
```diff
@@ -48,10 +48,109 @@
...
+Notebooks provide a nice environment for exploratory and pedagogical code development.
+Some reasons they are good for this type of work are:
+
+- They incorporate tab completion and easy access to help information
+- Documentation can reside with the code via markdown cells
+- Cells can be executed in arbitrary order, so the entire script doesn't need to be run to iterate on a particular part of the notebook
+- Code in cells can be flexible.  E.g. one cell can execute bash commands while the next cell contains python
+
+The same reasons notebooks perform well in exploratory scenarios are the reasons they are somewhat non-optimal for rigorous software engineering.
+A concrete example is that running cells in arbitrary order makes it easy to end up with a notebook in a state where it will not successfully complete when run sequentially.
```
Comment (leannep) [inline index.rst:75]: , or may complete successfully but give erroneous results if cells have not been run in the correct order.

File: index.rst [hunk 2]
```diff
@@ -48,10 +48,109 @@
...
+  
+These repositories will be ``eups`` packages so that we can use all the existing infrastructure.
+
+Notes on notebook review
+------------------------
+
+Notebooks should be reviewed by someone other than the original author.
+The review should look for several things:
+
+- The notebook should adhere to a standard templated format (not defined here).
```
Comment (leannep) [inline index.rst]: Is one defined elsewhere that could be linked to? Stack-club have developed a nice template format, could we adopt it?

File: index.rst [hunk 3]
```diff
@@ -48,10 +48,109 @@
...
+
+Have a concrete plan for what to do about notebooks that do not pass testing.
+At some point they should be moved back to ``notebooks/WIP``, but whether that is immediate or there is a grace period for the author to fix the notebook should be up to the repository owner.
+
+Whether notebooks are cleared before committing, will depend on the intentions of the team using the notebook repository.
+
+- Clearing outputs will reduce the overall size, make the diff cleaner, and makes the notebook clean for the next person to execute
+- Leaving outputs in the committed notebook will save time since the notebook does not need to be executed to see plots etc.
+  Leaving outputs also allows for robust testing since ``nbeval`` can compare outputs from the tested notebook match those of the input notebook.
+
```
Comment (leannep) [inline index.rst:148]: For DM testing, see DMTR-201 and  DMTR-161, a convention that has been adopted is to have a directory called 'notebooks' and a directory called 'executions'. The latter is used to store notebooks with outputs, if desired.

**Review by leannep — CHANGES_REQUESTED** [review]

File: index.rst (hunk 2 above)
Comment (SimonKrughoff) [reply index.rst]: I don't think it is templated, but I can look.

**Pull request merged**